File: my_cpp_pkg/src/number_reset.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/set_bool.hpp"

class numberResetNode : public rclcpp::Node 
{
public:
    numberResetNode() : Node("number_reset") 
    {
        thread1_ = std::thread(std::bind(&numberResetNode::callNumberReset, this, false));

    }

    void callNumberReset(bool msg_bool)
    {
        auto client = this->create_client<example_interfaces::srv::SetBool>("reset_number_count");

        while (!client->wait_for_service(std::chrono::seconds(1)))
        {
            RCLCPP_WARN(this->get_logger(), "Waiting for the server to be up...");
        }

        auto request = std::make_shared<example_interfaces::srv::SetBool::Request>();
        request->data = msg_bool;

        auto future = client->async_send_request(request);

        try 
        {
            auto response = future.get();
        
            // RCLCPP_INFO(this->get_logger(), "%d", (int)response->success);
            // RCLCPP_INFO(this->get_logger(), "%s", ((std::string)response->message).c_str());

            RCLCPP_INFO(this->get_logger(), "%d", response->success);
            RCLCPP_INFO(this->get_logger(), "%s", (response->message).c_str());

        }

        catch (const std::exception &e)
        {
            RCLCPP_ERROR(this->get_logger(), "Service Call Failed");
        }


    }

private:
    std::thread thread1_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<numberResetNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/number_reset.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from functools import partial
from example_interfaces.srv import SetBool


class NumberResetNode(Node):
    def __init__(self):
        super().__init__("reset_counter_client")
        self.call_reset_counter_server(False) # Actual function call

    def call_reset_counter_server(self, bool_val):
        client = self.create_client(SetBool, "reset_number_count")

        while not client.wait_for_service(1.0):
            self.get_logger().warn("Waiting for Server Reset Number Count...")

        request = SetBool.Request()
        request.data = bool_val

        future = client.call_async(request)
        future.add_done_callback(
            partial(self.callback_call_reset_number))

    def callback_call_reset_number(self, future):
        try:
            response = future.result()
            print(type(response.success))
            print(type(response.message))
            # self.get_logger().info(response.success)
            # self.get_logger().info(response.message)

        except Exception as e:
            self.get_logger().error("Service call failed %r" % (e,))


def main(args=None):
    rclpy.init(args=args)
    node = NumberResetNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
